js/src/provider/WalletLinkProvider.ts

```diff
@@ -940,16 +970,32 @@ export class WalletLinkProvider
   ): Promise<JSONRPCResponse> {
     const request = (params[0]) as AddEthereumChainParams;
 
+    if (request.rpcUrls?.length === 0) {
+      return { jsonrpc: '2.0', id: 0, error: { code: 2, message: `please pass in at least 1 rpcUrl` } };
+    }
+
     const chainIdNumber = parseInt(request.chainId, 16);
     const ethereumChain = EthereumChain.fromChainId(BigInt(chainIdNumber));
     if (ethereumChain === undefined) {
-      return { jsonrpc: '2.0', id: 0, error: { code: 2, message: `chainId ${request.chainId} not supported` } };
+      const success = await this.addEthereumChain(
```

Comment:
If the caller:
1. Adds new ethereum chain that isn't hardcoded in `EthereumChain.fromChainId`
2. Calls `wallet_switchEthereumChain`
The code will throw an error since `wallet_switchEthereumChain` will try to use `EthereumChain.fromChainId` on the unsupported chain.

Just want to clarify if this is the expected behavior or whether we should add support for it.

Reply:
For consistency, it looks like other wallets support calling `wallet_switchEthereumChain` on a newly added chain.

Reply:
[]()_**~~~~**_